design-proposals/helm-deployment/helmDeploymentRequirements.md

```diff
@@ -0,0 +1,999 @@
+
+  ```
+  helm upgrade --set snapshot.enable=true sample-esdk 
+  ```
+You can specify the '--values'/'-f' flag multiple times. The priority will be given to the last (right-most) file specified. For example, if both myvalues.yaml and override.yaml contained a key called 'Test', the value set in override.yaml would take precedence:  
+
+    `$ helm upgrade -f myvalues.yaml -f override.yaml sample-esdk`  
+
+
+- Roll back a release:  
+    `$ helm rollback sample-esdk <revision>`  
+
+- Delete a release:  
+    `$ helm delete sample-esdk`  
+
+
+
+### Use case View
+
+
+
+
+#### List of Typical Usecases
+
+
+ 
+
+### Development and Deployment Context
+
+#### Code
+NA
+#### Debug Model
+NA
+
+  
+#### Build & Package
+NA
+
+  #### Deployment
+
+//How to install and deploy the module in the system, hardware resource requirements etc. Any other network or such requirements..like client or http server needed etc…
+
+  
+### Execution View
+
+//During the run time, any specific aspects to be considered...like logging to be done for the module etc..It is not functional logs, it is specific to the module maintenance; OR Runtime replication or any such requirements to be considered during the design
+
+  
+## Sequence Diagrams
+
+//Provide the key control and data flow sequence diagrams here
+
+  
+## Design Alternatives and other notes
+
+//If you have any other ideas or alternate suggestions or notes which needs further analysis or later consideration, please add here
+  
+
+## Open Issues
+
+//All the open issues go here. Please track it else where to closure
+
+  
+## Design Requirements / Tasks
+
+//List of detailed tasks for this module go here. Based on all the design and analysis, please list all the tasks to be completed for the implementation and release of the module. If you are updating the overall task list or location, please provide the links or ids here...This is to get an overall consolidation of task items for this module
+
+## Scratchpad
+
+//All raw inputs or discussion points or etc can be added here
+https://github.com/jsafrane/community/blob/master/contributors/design-proposals/storage/container-storage-interface.md#recommended-mechanism-for-deploying-csi-drivers-on-kubernetes
+
+#### Multiarchitecture images with docker
+https://www.docker.com/blog/multi-arch-build-and-images-the-simple-way/
+
+https://github.com/docker-library/official-images#architectures-other-than-amd64
+
+https://medium.com/@artur.klauser/building-multi-architecture-docker-images-with-buildx-27d80f7e2408
+
+#### Dell powermax CSI Helm deployment
+https://github.com/dell/csi-powermax/tree/master/helm
+
+
+#### K8S Secrets
+https://kubernetes.io/docs/concepts/configuration/secret/
+
+#### Helm architecture
+https://helm.sh/docs/topics/architecture/
+
+#### Helm Chart hooks
+https://rafay.co/the-kubernetes-current/helm-chart-hooks-tutorial/
+
+
+**Open Points**
```

Comment:
IMO, open points can be discussed and resolved

Reply:
This is the scratchpad section. The open points do need to be discussed and resolved.

Reply:
Removed sections which are not relevant.